storage/range.go

```diff
@@ -774,6 +783,19 @@ func (r *Range) executeCmd(method string, args proto.Request, reply proto.Respon
 		return util.Errorf("unrecognized command %q", method)
 	}
 
+	// If we are applying a raft command, update the applied index.
```

Comment:
Why not combine this with the check below for err == nil? Put another if, first thing inside that err == nil block, which checks for index > 0.